cms/admin/utils.py

```diff
@@ -696,6 +699,51 @@ def save_model(self, request: HttpRequest, obj: models.Model, form: forms.Form,
             setattr(form._content_instance, self.grouper_field_name, obj)
             form._content_instance.save()
 
+    def get_search_fields(self, request):
+        """Return search fields for either grouper model or content model"""
+        content_search_fields = []
+        grouper_search_fields = []
+        for field_name in self.search_fields:
+            if field_name.startswith(f"{self.content_related_field}__") or field_name.startswith("content__"):
+                content_search_fields.append(re.sub(r"^([@^=]?)[^_]+__", r"\1", field_name))
+
+            else:
+                grouper_search_fields.append(field_name)
+
+        if getattr(self, "_content_fields", False):
+            return content_search_fields
+
+        return grouper_search_fields
+
+    def get_search_results(self, request, queryset, search_term):
+        # Reset _content_fields flag
+        self._content_fields = False
```

Comment:
```suggestion
        request._content_fields = False
```
I'm not entirely sure, but to make this more thread-safe, I'd add the flag to the `request` object. The admin object might be called from a different request while the flag is set.